few edits

stemangiola opened the pull request:
Just few comments:

-	The beauty of having a tidy dataset is that you can not think about subsetting information “We will next extract just the columns we need, sample, transcript, count.” For example if lateron we want to visualise ensembl ID after analyses we can easily because we have that column.  We can forget about it until we need it. Pretty convenient!

I have done nothing here

-	Similarly, I would not hard-filter abundant genes. All analyses will be done on abundant genes anyway (unless I am missing something). So we can filter for plotting for example but I don’t think we need to delete any transcript from the data set

I have done nothing here

-	For “We can also create box plots to check the distributions of the counts in the samples” after scaling, I think we could first show the densities similarly to before so to see the difference. Even better I think is really good to see the potentiality of tidy data is to gather(c(count, count_caled), c(“source”, “abundance)) %>% ggplot + geom_density + facet_wrap(~source). Facetting based on whether scaled or not, one next to the other (added to document for feedback)
-	drop_na, is not implemented in tidybulk yet and will strip the internals 😊 I replaced with filter()

Please review the changes and keep if you like.

stemangiola committed:
few edits

mblue9 commented on vignettes/tidytranscriptomics.Rmd:
```diff
@@ -302,6 +304,19 @@ counts_scaled
 
 After we run `scale_abundance()` we should see some columns have been added at the end. We have a column called `lowly_abundant` that indicates whether the gene has been filtered due to being lowly expressed. FALSE means the gene wasn’t filtered, TRUE means it was. The `count_scaled` column contains the scaled counts.
 
+We can now see the difference of abundance densities before and adfter scaling
```
```suggestion
We can now see the difference of abundance densities before and after scaling
```

mblue9 commented on vignettes/tidytranscriptomics.Rmd:
```diff
@@ -375,19 +390,15 @@ An alternative to MDS for examining relationships between samples is using hiera
 	      .value = count_scaled,
 	      annotation = c(dex, cell),
-	      transform = log1p
+	      transform = function(x) x %>% log1p %>% tidyHeatmap:::scale_robust()
```
this is a failure here:

```Quitting from lines 390-403 (tidytranscriptomics.Rmd) 
Error: processing vignette 'tidytranscriptomics.Rmd' failed with diagnostics:
object 'scale_robust' not found
--- failed re-building ‘tidytranscriptomics.Rmd’

is scale_robust only available in tidybulk dev? we're supposed to be able to use bioc devel so is there a way to get it into that?

stemangiola replied:
I will make it exported from tidyHeatmap and update CRAN. Strange that gives error to you. Aa maybe because you have the cran version of course. I will update CRAN don't worry

stemangiola replied:
In the meanwhile could you please update tidyHeatmap from github?

mblue9 replied:
I could but that won't stop the failure, because Github actions is automatically installing the packages from CRAN and Bioc devel, you can see the details below if you haven't already
<img width="925" alt="Screen Shot 2020-05-30 at 7 17 50 pm" src="https://user-images.githubusercontent.com/5591821/83324736-b7340080-a2aa-11ea-805c-40cf27ca3043.png">

<img width="1419" alt="Screen Shot 2020-05-30 at 7 17 21 pm" src="https://user-images.githubusercontent.com/5591821/83324754-c6b34980-a2aa-11ea-964d-f3d52a20ad8d.png">

If it fails it means the changes don't get pushed to our website that the Github actions are automatically building here https://stemangiola.github.io/bioc_2020_tidytranscriptomics/articles/tidytranscriptomics.html

So if the change will get to CRAN soon we could wait otherwise if it will take a while (more than a day?) it's probably better to remove that scale_robust so we can add the other changes, then add the scale_robust when it's in CRAN

stemangiola replied:
I see. The CRAN tidyHeatmap it does scale automatically. I see we were seing to different plots because you have the CRAN version. I as seing it not scaled and so very ugly.

In this case discart my change!

mblue9 replied:
ok will do 😄

mblue9 reviewed (commented):
sounds good, let's add your changes and see how it looks

stemangiola committed:
Update vignettes/tidytranscriptomics.Rmd

Co-authored-by: mblue9 <maria.doyle@petermac.org>

mblue9 commented on vignettes/tidytranscriptomics.Rmd:
```diff
@@ -375,19 +390,15 @@ An alternative to MDS for examining relationships between samples is using hiera
 	      .value = count_scaled,
 	      annotation = c(dex, cell),
-	      transform = log1p
+	      transform = function(x) x %>% log1p %>% tidyHeatmap:::scale_robust()
```
```suggestion
	      transform = log1p
```
So I think if you accept this suggested change to revert this PR should hopefully pass

stemangiola committed:
Update vignettes/tidytranscriptomics.Rmd

Co-authored-by: mblue9 <maria.doyle@petermac.org>

mblue9 commented:
Nice! all green now 👍 

I could merge now? or you could add those other edits you suggested. I could make them but I'm not exactly sure what changes should be made for `Similarly, I would not hard-filter abundant genes. All analyses will be done on abundant genes anyway (unless I am missing something). So we can filter for plotting for example but I don’t think we need to delete any transcript from the data set`

Could you make the changes so I can see what you mean and then I could comment?

stemangiola committed:
don't need to sekect information unless decrease incredibly the size of the huge dataset. It's good to not have to worry.

stemangiola committed:
avoid to drop information from the source data set filtering lowly transcribed genes

stemangiola committed:
type

Merge branch 'few-edits' of https://github.com/stemangiola/bioc_2020_tidytranscriptomics into few-edits

Conflicts:
	vignettes/tidytranscriptomics.Rmd

stemangiola commented:
> Could you make the changes so I can see what you mean and then I could comment?

Done, have a look

mblue9 commented on vignettes/tidytranscriptomics.Rmd:
```diff
@@ -254,33 +245,34 @@ The bar plots show us there are 15-30 million counts per sample.
 Genes with very low counts across all libraries provide little evidence for differential expression and they interfere with some of the statistical approximations that are used later in the pipeline. They also add to the multiple testing burden when estimating false discovery rates, reducing power to detect differentially expressed genes. These genes should be filtered out prior to further analys ...
 
-tidybulk can automatically filter out lowly expressed genes. It uses the edgeR filterByExpr function described [here](https://f1000research.com/articles/5-1408). This will keep genes with ~10 counts in a minimum number of samples, the number of the samples in the smallest group. In this dataset the smallest group size is 2 samples. tidybulk performs this filtering in the functions we will use `sc ...
+With tidybulk, it is not really needed to explicitly filter lowly transcribed genes as all calulations (e.g., scaling, removal unwanted variation, differential expression testing) are done on abundantly transcribed genes; although in case of scaling, it is apllied back on all transcripts.
```
```suggestion
With tidybulk, it is not really necessary to explicitly filter lowly transcribed genes, as all calculations (e.g., scaling, removal of unwanted variation, differential expression testing) are performed on abundantly transcribed genes; although in case of scaling, the scaling is applied back to all genes in the dataset.
```

mblue9 commented on vignettes/tidytranscriptomics.Rmd:
````diff
@@ -254,33 +245,34 @@ The bar plots show us there are 15-30 million counts per sample.
 ```{r}
-# Filtering out lowly abundant genes 
-counts_filt <- counts_aggr %>% 
+# Take a look to only highly abundant genes 
````
```suggestion
# Take a look at the abundant genes
```

mblue9 committed:
Update vignettes/tidytranscriptomics.Rmd

mblue9 committed:
Update vignettes/tidytranscriptomics.Rmd

mblue9 commented:
Ok I made a couple of small edits and I'll merge now to see what it looks like. If we need to make more changes we can do that later.
